test_runner/regress/test_physical_replication.py

```diff
@@ -42,3 +42,39 @@ def test_physical_replication(neon_simple_env: NeonEnv):
                                 s_cur.execute(
                                     "select * from t where pk=%s", (random.randrange(1, 2 * pk),)
                                 )
+
+
+def test_physical_replication_config_mismatch(neon_simple_env: NeonEnv):
+    env = neon_simple_env
```

Comment:
please add a test description

Reply:
Done

Reply:
> Let's add tests for overrunning max_prepared_transactions and max_locks_per_transactions too. I.e. a test that creates many prepared transactions in primary, and a test that acquires a lot of AccessExclusiveLocks in the primary.

I have added test for `max_prepared_transactions` and `too many known xids` proposed by @MMeent.
On both cases walreveiver is stopped with fatal error.

I failed to cause replica crash because of locks exhaustion (but didn't change `max_locks_per_transactions`, just `max_connections`). Do you wan yet another negative test or two is enough?

Reply:
I have added the test